rcl/src/rcl/arguments.c

```diff
@@ -335,16 +339,21 @@ rcl_parse_arguments(
             i += 1;  // Skip flag here, for loop will skip rule.
             continue;
           }
-          RCUTILS_LOG_DEBUG_NAMED(ROS_PACKAGE_NAME,
-            "Couldn't parse arg %d (%s) as remap rule. Error: %s", i + 1, argv[i + 1],
-            rcl_get_error_string().str);
+          rcl_error_string_t prev_error_string = rcl_get_error_string();
           rcl_reset_error();
+          RCL_SET_ERROR_MSG_WITH_FORMAT_STRING(
+            "Couldn't parse remap rule: '%s %s'. Error: %s", argv[i], argv[i + 1],
+            prev_error_string.str);
         } else {
-          RCUTILS_LOG_DEBUG_NAMED(ROS_PACKAGE_NAME,
-            "Couldn't parse arg %d (%s) as remap rule flag. No rule found.\n",
-            i, argv[i]);
+          RCL_SET_ERROR_MSG_WITH_FORMAT_STRING(
+            "Couldn't parse trailing remap rule: '%s'. No rule found.", argv[i]);
         }
+        ret = RCL_RET_INVALID_ROS_ARGS;
+        goto fail;
       }
+      RCUTILS_LOG_DEBUG_NAMED(ROS_PACKAGE_NAME,
+        "Couldn't parse arg %d (%s) as remap rule. Not one of the %s or %s flags.", i,
```

Comment:
same about `Not one of '%s' or '%s' flags.`

Reply:
Same as above.